v2/kafka-to-kafka/src/main/java/com/google/cloud/teleport/v2/templates/ConsumerProperties.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (C) 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.templates;
+
+import com.google.cloud.ServiceOptions;
+import com.google.cloud.secretmanager.v1.AccessSecretVersionResponse;
+import com.google.cloud.secretmanager.v1.SecretManagerServiceClient;
+import com.google.cloud.secretmanager.v1.SecretVersionName;
+import com.google.cloud.teleport.v2.options.KafkaToKafkaOptions;
+import com.google.common.collect.ImmutableMap;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.kafka.clients.CommonClientConfigs;
+import org.apache.kafka.common.config.SaslConfigs;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+final class ConsumerProperties {
+
+  private static final Logger LOGG = LoggerFactory.getLogger(ConsumerProperties.class);
+
+  public static List<String> accessSecretVersion(String projectId, KafkaToKafkaOptions options) {
```

Comment:
This code seems to be duplicated between ConsumerProperties and ProducerProperties, consider extracting it into a separate (parent?) class or combining Consumer/Producer properties somehow.